cob_image_flip/ros/src/image_flip.cpp

```diff
@@ -192,6 +192,70 @@ bool ImageFlip::convertImageMessageToMat(const sensor_msgs::Image::ConstPtr& ima
 }
 
 
+template <typename T> void ImageFlip::atForDiffImTypes(cv::Mat& image, int row_index, int column_index, double value, bool replace_prev_value)
+{
+    // default case is that the current value of the matrix-element is overwritten
+    if (replace_prev_value)
+    {
+        image.at<T>(row_index, column_index) = T(value);
+    // for adding a value to the current value use replace_prev_value = false
+    }else
+    {
+    image.at<T>(row_index, column_index) += T(value);
+    }
+}
+
+void ImageFlip::atForDiffImTypes(cv::Mat& image, int row_index, int column_index, double value, bool replace_prev_value)
+{
+    switch(image.type() % 8)
+    {
+    // image.type() // 8 is related to the number of channels which doesn't matter here
+    // for a table about the 28 different opencv-mat-types please refer to:
+    // http://ninghang.blogspot.com/2012/11/list-of-mat-type-in-opencv.html
+
+    case 0: atForDiffImTypes<unsigned char>(image, row_index, column_index, value, replace_prev_value);  break;
+    case 1: atForDiffImTypes<signed char>(image, row_index, column_index, value, replace_prev_value);  break;
+    case 2: atForDiffImTypes<unsigned short>(image, row_index, column_index, value, replace_prev_value);  break;
+    case 3: atForDiffImTypes<signed short>(image, row_index, column_index, value, replace_prev_value);  break;
+    case 4: atForDiffImTypes<int>(image, row_index, column_index, value, replace_prev_value);  break;
+    case 5: atForDiffImTypes<float>(image, row_index, column_index, value, replace_prev_value);  break;
+    case 6: atForDiffImTypes<double>(image, row_index, column_index, value, replace_prev_value);  break;
+
+    default: ROS_ERROR("ImageFlip::imageCallback: Error: Strange OpenCV-Mat-Type.\n"); break;
+    }
+}
+
+template <typename T> double ImageFlip::atForDiffImTypesReturn(cv::Mat& image, int row_index, int column_index)
```

Comment:
Maybe it would make sense to return T instead of double because the type conversion to double or something else could be done by the outside function?